auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -512,6 +576,50 @@ interface CredentialParseFunction {
         void parseCredential(String algorithm, String format, String body) throws RealmUnavailableException, XMLStreamException;
     }
 
+    /**
+     * Re-generate the signatures for all the identities in this realm.
+     * This method is intended to be called after updating the key pair used by this realm.
+     *
+     * @throws RealmUnavailableException Thrown if unable to obtain the realm identity iterator or dispose and close the realm
+     */
+    public void updateRealmKeyPair() throws RealmUnavailableException {
+        if (! hasIntegrityEnabled()) {
+            throw ElytronMessages.log.integrityNotEnabled(root.toString());
+        }
+        ModifiableRealmIdentityIterator realmIterator = this.getRealmIdentityIterator();
+        while (realmIterator.hasNext()) {
+            Identity identity = (Identity) realmIterator.next();
+            try{
+                identity.writeDigitalSignature(String.valueOf(identity.path), identity.name);
+            } finally {
+                identity.dispose();
+            }
+        }
+        realmIterator.close();
+    }
+
+    /**
+     * Verify the integrity of each identity file in this realm.
+     * @return {@code true} if the integrity of all the identity files in the realm is successfully verified and {@code false} otherwise
+     *
+     */
+    public boolean verifyRealmIntegrity() throws RealmUnavailableException {
+        if (! hasIntegrityEnabled()) {
+            throw ElytronMessages.log.integrityNotEnabled(root.toString());
+        }
+        ModifiableRealmIdentityIterator realmIterator = this.getRealmIdentityIterator();
+        while (realmIterator.hasNext()) {
+            Identity identity = (Identity) realmIterator.next();
+            if(! identity.verifyIntegrity()) {
+                identity.dispose();
```

Comment:
@fjuma @Ashpan Do we want to log which identity failed the integrity check?

Reply:
Yes that's a good point.

Reply:
@Ashpan Ah yes, sorry I missed that. Although it could still happen that the `validatePrincipalName(path.toString(), name)` fails instead of the digital signature, and then no identity is logged right?